changelog/db.changelog-1.1.xml

```diff
@@ -0,0 +1,131 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-3.9.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.9.xsd">
+ <changeSet author="Marc Templeton" id="2">
+    <tagDatabase tag="v1.0.0" />
+  </changeSet>
+  <changeSet author="Marc Templeton" id="3">
+    <createTable tableName="paymentRequests">
+      <column name="paymentRequestId" type="INTEGER" autoIncrement="true">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="paymentRequestsId_pkey" />
+      </column>
+      <column name="agreementNumber" type="VARCHAR(20)" />
+      <column name="contractNumber" type="VARCHAR(10)" />
+      <column name="correlationId" type="VARCHAR(36)" /> <!-- interesting 1.23 rather tnan uuid -->
+      <column name="currency" type="VARCHAR(3)" />
+      <column name="deliveryBody" type="VARCHAR(4)" />
+      <column name="dueDate" type="VARCHAR(10)" />
+      <column name="invoiceNumber" type="VARCHAR(30)" />
+      <column name="marketingYear" type="INTEGER" />
+      <column name="referenceId" type="UUID" /> <!-- interesting 1.23 -->
+      <column name="schedule" type="VARCHAR(3)" />
+      <column name="sitiAgriInvoiceNumber" type="VARCHAR(20)" />
+      <column name="submitted" type="TIMESTAMP" />
+      <column name="value" type="INTEGER" />
+    </createTable>
+  </changeSet>
+  <changeSet author="Marc Templeton" id="4">
+    <createTable tableName="settlements">
+      <column name="settlementId" type="INTEGER" autoIncrement="true">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="settlementsId_pkey" />
+      </column>
+      <column name="paymentRequestId" type="INTEGER" >
+        <constraints nullable="false" />
+      </column>
+      <column name="detail" type="VARCHAR(255)" />
+      <column name="invalid" type="BOOLEAN" defaultValue="false" />
+      <column name="generated" type="BOOLEAN" defaultValue="false" />
+      <column name="frn" type="BIGINT" />
+      <column name="invoiceNumber" type="VARCHAR(30)" />
+      <column name="ledger" type="VARCHAR(2)" />
+      <column name="reference" type="VARCHAR(9)" /> <!-- based on PY1234567 -->
+      <column name="settled" type="BOOLEAN" defaultValue="false" />
+      <column name="settlementDate" type="TIMESTAMP" /> <!-- interesting in pay-pr this (named settled) is used for above bool -->
```

Comment:
In `ffc-pay-processing` originally we only had one field `settled` as a boolean that we used to determine whether a payment request was settled in DAX.

However, we had to change it as we are dealing with quarterly payments.  So instead of one response for an invoice number to indicate it is fully settled, we actually get four responses, each with an incrementing total settled value.

So we dropped the `settled` column and replaced it with `settledValue` and `lastSettlementDate`.  The settled value is the total we have settled so far against that invoice number whilst the last settlement date is the date of the latest settlement.  This is used to handle scenarios where we get duplicate notifications for settlements due to either BACS rejections where earlier settlements will be resent by DAX or even duplicate messages in our own infrastructure.

So the date isn't being used (entirely) as a boolean.

As for what we should do here as we will have a similar scenario.  We don't want duplicate settlements to trigger a duplicate statement.  So do we either check before saving using this date, similar to processing, or do we always save the settlement and use logic to filter out duplicate or redundant settlements.

I think the answer to that question lies in what do we do about statements and BACS rejections.  Do they need another refreshed statement or are they ignored.

Reply:
I'm going to leave as-is due to the fact this is how we've handled these kinds of scenarios in other services